Sources_v3/WebSocketClient/WebSocketClient.swift

```diff
@@ -353,3 +353,25 @@ protocol BackgroundTaskScheduler {
 }
 
 extension UIApplication: BackgroundTaskScheduler {}
+
+struct HealthCheckMiddleware: EventMiddleware {
+    private(set) weak var webSocketClient: WebSocketClient?
+
+    func handle(event: Event, completion: @escaping (Event?) -> Void) {
+        guard let healthCheckEvent = event as? HealthCheckEvent else {
+            // Do nothing and forward the event
+            completion(event)
+            return
+        }
+        
+        if let webSocketClient = webSocketClient {
+            webSocketClient.pingController.pongRecieved()
+            if webSocketClient.connectionState.isConnected == false {
+                webSocketClient.connectionState = .connected(connectionId: healthCheckEvent.connectionId)
+            }
+        }
+        
+        // Stop the middleware chain
```

Comment:
nit: It actually doesn't stop the chain. Maybe something like `// Don't forward the event` is more on spot.

Reply:
fixed in c7b85be94bb9bec60b9efe17f7f2c4921730f77b